cnd/src/http_api/routes/rfc003/handlers/post_swap.rs

```diff
@@ -119,8 +131,10 @@ where
 
             match response {
                 Ok(accept) => {
-                    let state = State::accepted(swap_request.clone(), accept.clone(), seed);
+                    let state = State::accepted(swap_request.clone(), accept, seed);
                     StateStore::insert(&dependencies, id, state.clone());
+                    SaveMessage::save_message(&dependencies, accept)
+                        .expect("failed to save message to db");
```

Comment:
:question: Is it intentional to panic if this fails?

Reply:
It was a trade-off we had to make:

We wanted to know if it fails but since this is inside a future, we can't quite handle the error properly :(